core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala

```diff
@@ -364,10 +367,13 @@ abstract class BaseProducerSendTest extends KafkaServerTestHarness {
     // Trying to send a record to a partition beyond topic's partition range before adding the partition should fail.
     val partition1 = 1
     try {
-      producer.send(new ProducerRecord(topic, partition1, null, "value".getBytes(StandardCharsets.UTF_8)))
+      producer.send(new ProducerRecord(topic, partition1, null, "value".getBytes(StandardCharsets.UTF_8))).get()
       fail("Should not allow sending a record to a partition not present in the metadata")
     } catch {
-      case _: KafkaException => // this is ok
+      case e: ExecutionException => e.getCause match {
+        case _: TimeoutException => // this is ok
+        case _ => fail("Sending to a partition not present in the metadata should result in a TimeoutException")
```

Comment:
Maybe we can just raise the exception? Otherwise we'll lose the trace in the output. Similarly in `ProducerFailureHandlingTest`.